libnetwork_test.go

```diff
@@ -0,0 +1,349 @@
+				// I would like to verify that the correct pools were returned to Docker but it doesn't let us observe that information - https://github.com/docker/docker/issues/28567
+			})
+			It("creates a network with a subnet", func() {
+				session := DockerSession("docker network create success$RANDOM --subnet 192.169.0.0/16 -d calico --ipam-driver calico-ipam")
+				Eventually(session).Should(Exit(0))
+			})
+
+			PIt("creates a network with IPv6", func() { //TODO - IPv6 support
+			})
+			PIt("creates a network with IPv6 from a specific subnet", func() {
+			})
+
+			//TODO - allow multiple networks from the same pool
+		})
+	})
+	Describe("docker network rm", func() {
+		// No options and no side effects
+	})
+	Describe("docker network connect", func() {
+		// TODO
+		// Usage:	docker network connect [OPTIONS] NETWORK CONTAINER
+		//
+		//Connect a container to a network
+		//
+		//Options:
+		//      --alias value           Add network-scoped alias for the container (default [])
+		//      --help                  Print usage
+		//      --ip string             IP Address
+		//      --ip6 string            IPv6 Address
+		//      --link value            Add link to another container (default [])
+		//      --link-local-ip value   Add a link-local address for the container (default [])
+		//
+	})
+	Describe("docker network disconnect", func() {
+		// TODO - no significant options but we should observe the veth going and the endpoint removed from etcd
+	})
+
+	//docker create/run
+	// create - doesn't have any network interactions until the container is started
+	// run - Run a container, check the following
+	//		- etcd contains correct info
+	//		- host namespace contains the right veth with the right info
+	//    - container - container the right routes and interface
+	//	run can have the following variations -
+	//    --mac-address
+	//    --link-local-ip
+	//    --ip and --ip6
+	Describe("docker run", func() {
+		var name string
+		BeforeEach(func() {
+			name = fmt.Sprintf("run%d", rand.Uint32())
+			DockerString(fmt.Sprintf("docker network create %s -d calico --ipam-driver calico-ipam", name))
+		})
+		AfterEach(func() {
+			DockerString(fmt.Sprintf("docker network rm %s", name))
+		})
+
+		It("creates a container on a network  and checks all assertions", func() {
+			// Create a container that will just sit in the background
+			DockerString(fmt.Sprintf("docker run --net %s -tid --name %s busybox", name, name))
+
+			// Gather information for assertions
+			docker_endpoint := GetDockerEndpoint(name, name)
+			ip := docker_endpoint.IPAddress
+			mac := docker_endpoint.MacAddress
+			endpoint_id := docker_endpoint.EndpointID
+			interface_name := "cali" + endpoint_id[:mathutils.MinInt(11, len(endpoint_id))]
+
+			// Check that the endpoint is created in etcd
+			etcd_endpoint := GetEtcdString(fmt.Sprintf("/calico/v1/host/test/workload/libnetwork/libnetwork/endpoint/%s", endpoint_id))
```

Comment:
I'd prefer to use the libcalico-go library here... Not convinced it should block check in tho.